packages/react-button/src/components/Button/useButton.ts

```diff
@@ -6,8 +7,33 @@ import { ButtonProps, ButtonState } from './Button.types';
  * @param props - Button props to derive state from.
  */
 export const useButton = (props: ButtonProps): ButtonState => {
+  const { as, onClick, onKeyDown: onButtonKeyDown } = props;
+
+  const onKeyDown = (ev: React.KeyboardEvent<HTMLButtonElement> & React.MouseEvent<HTMLButtonElement>) => {
+    if (onButtonKeyDown) {
+      onButtonKeyDown(ev);
+    }
+
+    if (onClick) {
+      const eventCode = getCode(ev);
+      if (eventCode === EnterKey || eventCode === SpacebarKey) {
+        onClick(ev);
+      }
+    }
+  };
+
+  const isButtonRootType = !as || as === 'button';
+  const rootTypeProps = isButtonRootType
+    ? {}
+    : {
+        onKeyDown,
```

Comment:
would it be possible to use v0 behaviors in some form? another use case - if this is composed into a toggle button, aria-pressed needs to be applied. this logic can get hairy and it would be good to have it externalized/centralized so that the components are not polluted with it

Reply:
@jurokapsiar I'm making use of it now, please take a look again.